include/envoy/server/worker.h

```diff
@@ -78,7 +78,7 @@ class Worker {
    * Stop all listeners from accepting new connections. This is used for server draining.
    * TODO(mattklein123): Same comment about the addition of a completion as stopListener().
    */
-  virtual void stopListeners() PURE;
+  virtual void stopListeners(std::function<void()> completion) PURE;
```

Comment:
Note this is going to conflict with https://github.com/envoyproxy/envoy/pull/8415. Let's get that merged first. (I would recommend locally rebasing this change on top of that to get a head start). cc @ramaraochavali 

nit: doc comments need updating.